src/components/ApplicationGroup/Details/EnvironmentOverview/EnvironmentOverview.tsx

```diff
@@ -126,6 +139,35 @@ export default function EnvironmentOverview({
     const getDeploymentHistoryLink = (appId: number, pipelineId: number) =>
         `/application-group/${envId}/cd-details/${appId}/${pipelineId}/`
 
+    const sortByApplication = () => {
+        handleSorting(EnvironmentOverviewSortableKeys.application)
+    }
+
+    const sortByDeployedAt = () => {
+        handleSorting(EnvironmentOverviewSortableKeys.deployedAt)
+    }
+
+    const sortAppListData = () => {
+        setAppListData((_appListData) => ({
+            ..._appListData,
+            appInfoList: _appListData.appInfoList.sort((a, b) => {
+                if (sortBy === EnvironmentOverviewSortableKeys.deployedAt) {
+                    return handleRelativeDateSorting(a.lastDeployed, b.lastDeployed, sortOrder)
+                }
+
+                return sortOrder === SortingOrder.ASC
+                    ? a.application.localeCompare(b.application)
+                    : b.application.localeCompare(a.application)
+            }),
+        }))
+    }
+
+    useEffect(() => {
+        if (appListData) {
+            sortAppListData()
+        }
+    }, [sortBy, sortOrder, JSON.stringify(appListData)])
```

Comment:
Instead of adding JSON.stringify(appListData) as dependency we can pass the applistData to sortAppListData and call teh same from line no 198